### PR TITLE
RELATED: RAIL-3446 impelment all filter-related commands (incl. tests)

### PR DESCRIPTION
**Pull request by no23reason**
As part of this, mockingbird was extended to support getCommonAttributes calls.

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4049/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 4s
- gooddata-ui-sdk-unit-tests : SKIPPED

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4050/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5682/ : SUCCESS in 19m 02s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/utils.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+export function* getAttributeFilterIndexById(filterLocalId: string): SagaIterator<number> {
+    const allFilters: ReturnType<typeof selectFilterContextAttributeFilters> = yield select(
+        selectFilterContextAttributeFilters,
+    );
+
+    return allFilters.findIndex((filter) => filter.attributeFilter.localIdentifier === filterLocalId);
+}
+
+export function* getAttributeFilterByDisplayForm(
```
[the remark this answers is not in the document]
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/utils.ts]: True, reworked.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/addAttributeFilterHandler.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        filterContextActions.addAttributeFilter({
+            displayForm,
+            index,
+            initialIsNegativeSelection,
+            initialSelection,
+            parentFilters,
+        }),
+    );
+
+    const addedFilter: SagaReturnType<typeof getAttributeFilterByDisplayForm> = yield call(
```
[the remark this answers is not in the document]
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/addAttributeFilterHandler.ts]: True, reworked.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1451

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1180/ : SUCCESS in 1m 12s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1454/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/201/ : SUCCESS in 6m 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5696/ : SUCCESS in 16m 37s

**Pull request merged**